priorityq.py

```diff
@@ -60,10 +61,25 @@ def __init__(self, iterable=()):
         args:
             iterable: an optional iterable to add to the priority queue. Items
                       added this way will be given a priority of None.
+
+        each item inside iterable can be either:
+        * A QNode object
+        * A container with value, priority
+        * A non-iterable value
+
         """
         self.heap = BinaryHeap(iterable=())
         for item in iterable:
-            self.insert(item)
+            try:
+                is_container = len(item) == 2
```

Comment:
Understood. Then why don't we just eliminate "A non-iterable value" and require all non-QNode objects to be passed in as a container? It will simplify and eliminate unintended effects like described above.

So, each item inside iterable can be either:
* a QNode object
* a *tuple* with value, priority

I think we can be explicit here.